### PR TITLE
timm upgrade 0.6.12

### PR DESCRIPTION
**Pull request by VadimLevin**
This patch resolves compatibility issues after timm upgrade to 0.6.12 release.

Unfortunately, RegNet in 0.6.12 uses RegNetCfg dataclass instead of plain dict as cfg parameter ([related commit](https://github.com/rwightman/pytorch-image-models/commit/ab49d275de8a9c344aea086fd86d04c4cabb6098)).
This requires min Python version change to 3.7 (when dataclass module was introduced).

Related [issue 620](https://github.com/qubvel/segmentation_models.pytorch/issues/620)

### CODE REVIEW

**Comment by qubvel** [comment]
Hi Vadim, 
Thanks for your contribution!
I was really thinking about deprecating `timm-` encoders in favor of `tu-` enoders to make updates smooth, but seems like your solve the issue!

**Comment by qubvel** [comment]
Need to update Python version in CI 
https://github.com/qubvel/segmentation_models.pytorch/blob/5ab92dbc056404129f604a0c2c37794afaf96228/.github/workflows/pypi.yml#L15

File: segmentation_models_pytorch/encoders/timm_efficientnet.py [hunk 1]
```diff
@@ -105,7 +105,7 @@ def __init__(self, stage_idxs, out_channels, depth=5, **kwargs):
     def get_stages(self):
         return [
             nn.Identity(),
-            nn.Sequential(self.conv_stem, self.bn1, self.act1),
+            nn.Sequential(self.conv_stem, self.bn1),
```
[the remark this answers is not in the document]
Comment (VadimLevin) [reply segmentation_models_pytorch/encoders/timm_efficientnet.py:108]: Not gone away, it just merged with a batch norm.
```python
 norm_act_layer = get_norm_act_layer(norm_layer, act_layer)
# ...
 self.bn1 = norm_act_layer(stem_size, inplace=True)
```
[`get_norm_act_layer`](https://github.com/rwightman/pytorch-image-models/blob/ce4d3485b690837ba4e1cb4e0e6c4ed415e36cea/timm/models/layers/create_norm_act.py#L52)

**Comment by a99user** [comment]
@qubvel, could you reveal some details a bit regrading mentioned `tu-` encoders, please?

**Comment by qubvel** [comment]
Both `tu-` and `timm-` are models from `timm` (see timm universal). However, `tu-` encoders are not rewritten for this repo structure and are slightly less optimized for some features. 
For example, 
 - when you use encoder as `timm-` and `depth=4`, then images are passed **only** through **required** blocks
 - when you use `tu-` encoder  then images are passed through **all** blocks, no matter what depth is specified (and only required features are used)
 
 But, you have more options using `tu-` encores, almost any model from timm

**Comment by VadimLevin** [comment]
It looks like [PR-694](https://github.com/qubvel/segmentation_models.pytorch/pull/694) broke formatting

**Comment by qubvel** [comment]
Thanks a lot, merged!
